workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerHeartbeat/SendPlayerHeartbeatRequestSystem.cs

```diff
@@ -23,25 +24,20 @@ public class SendPlayerHeartbeatRequestSystem : ComponentSystem
     {
         private float timeOfNextHeartbeat = Time.time + PlayerLifecycleConfig.PlayerHeartbeatIntervalSeconds;
         private ComponentGroup group;
+        private CommandSystem commandSystem;
 
         protected override void OnCreateManager()
         {
             base.OnCreateManager();
 
-            var query = new EntityArchetypeQuery
-            {
-                All = new[]
-                {
-                    ComponentType.Create<PlayerHeartbeatClient.CommandSenders.PlayerHeartbeat>(),
-                    ComponentType.ReadOnly<Authoritative<PlayerHeartbeatServer.Component>>(),
-                    ComponentType.ReadOnly<HeartbeatData>(),
-                    ComponentType.ReadOnly<SpatialEntityId>(),
-                },
-                Any = Array.Empty<ComponentType>(),
-                None = Array.Empty<ComponentType>()
-            };
+            group = GetComponentGroup(
+                ComponentType.ReadOnly<PlayerHeartbeatServer.ComponentAuthority>(),
+                ComponentType.Create<HeartbeatData>(),
+                ComponentType.ReadOnly<SpatialEntityId>()
+            );
+            group.SetFilter(new PlayerHeartbeatServer.ComponentAuthority(true));
```

Comment:
I didn't know you could filter in `OnCreateManager`

Reply:
Neither did I. Have we verified that this works?

Reply:
Fixed the filtering in the last commit. It does work, just not with the ArchetypeChunk API.

Reply:
Filtering doesn't work with the chunk API anywhere though. The question is if it works on OnCreateManager. Looking at the ChangeFilter stuff it looks like that has to be done each time you get query the group.

Reply:
The code was changed to use the proper API, and confirmed to work with the filter set in OnCreateManager.